field_utils/FieldCondensedArray_private.F90

```diff
@@ -0,0 +1,42 @@
+#include "MAPL_Generic.h"
+module mapl3g_FieldCondensedArray_private
+
+#include "MAPL_Generic.h"
+   use MAPL_ExceptionHandling
+   implicit none
+
+   private
+   public :: get_array_shape
+
+contains
+
+   function get_array_shape(gridToFieldMap, localElementCount, vert_dims, rc) &
+         &result(array_shape)
+      integer :: array_shape(3)
+      integer, intent(in) :: gridToFieldMap(:)
+      integer, intent(in) :: localElementCount(:)
+      integer, optional, intent(in) :: vert_dims(:)
```

Comment:
Why is this optional?   Maybe it will be more obvious later in the code.

Reply:
The `get_array_shape` in `FieldPointerUtilities.F90` always calls the private function with an integer array, now, even when there is no vertical dimension. So this no longer needs to be optional.